src/epstats/toolkit/testing/test_dao.py

```diff
@@ -25,15 +26,21 @@ def get_agg_goals(self, experiment: Experiment) -> pd.DataFrame:
         # We take min from date_to and date_for in case we evaluate historical experiment where date_for >> date_to
         # already.
         if experiment.date_from is not None and experiment.date_to is not None:
-            return goals[
+            goals = goals[
                 (goals.date >= experiment.date_from.strftime("%Y-%m-%d"))
                 & (goals.date <= experiment.date_to.strftime("%Y-%m-%d"))
                 & (goals.date <= experiment.date_for.strftime("%Y-%m-%d"))
             ]
         if experiment.date_from is not None:
-            return goals[goals.date >= experiment.date_from.strftime("%Y-%m-%d")]
+            goals = goals[goals.date >= experiment.date_from.strftime("%Y-%m-%d")]
         if experiment.date_to is not None:
-            return goals[goals.date <= experiment.date_to.strftime("%Y-%m-%d")]
+            goals = goals[goals.date <= experiment.date_to.strftime("%Y-%m-%d")]
+
+        for f in experiment.filters:
+            if f.scope == FilterScope.exposure:
+                goals = goals[(goals.goal != "exposure") | (goals[f.dimension].isin(f.value))]
+            if f.scope == FilterScope.goal:
+                goals = goals[(goals.goal == "exposure") | (goals[f.dimension].isin(f.value))]
```

Comment:
I'm not sure if I got this:
If you filter "on exposure", then you want to save only relevant exposures? And vice versa for filtering "on goal"...?

Reply:
If `scope == 'exposure'`, we keep all goals but `exposure` unfiltered and we filter `exposure` goals. If `scope == 'goal'`, we keep all `exposure`s unfiltered and we filter all goals but `exposure`s.